python_modules/dagster/dagster/_core/definitions/result.py

```diff
@@ -62,3 +55,32 @@ def check_result_named(self, check_name: str) -> AssetCheckResult:
                 return check_result
 
         check.failed(f"Could not find check result named {check_name}")
+
+
+class MaterializeResult(AssetResult):
+    """An object representing a successful materialization of an asset. These can be returned from
+    @asset and @multi_asset decorated functions to pass metadata or specify specific assets were
+    materialized.
+
+    Attributes:
+        asset_key (Optional[AssetKey]): Optional in @asset, required in @multi_asset to discern which asset this refers to.
+        metadata (Optional[MetadataUserInput]): Metadata to record with the corresponding AssetMaterialization event.
+        check_results (Optional[Sequence[AssetCheckResult]]): Check results to record with the
+            corresponding AssetMaterialization event.
+        data_version (Optional[DataVersion]): The data version of the asset that was observed.
+    """
+
+
+@experimental
+class ObserveResult(AssetResult):
+    """An object representing a successful observation of an asset. These can be returned from
```

Comment:
These comments aren't accurate, right? (I know that in the future they may become accurate if we choose to go the `execution_type` route.

Reply:
The comments are accurate, with the caveat that you need a special metadata key/value pair (setting execution type to OBSERVATION) for it to work. I added some clarification in the docstring.